src/shared/Core/Authentication/OAuth/OAuth2Client.cs

```diff
@@ -383,10 +386,10 @@ protected Exception CreateExceptionFromResponse(string json)
         {
             if (TryCreateExceptionFromResponse(json, out OAuth2Exception exception))
             {
-                return exception;
+                return new Trace2OAuth2Exception(_trace2, exception.Message);

```

Comment:
This one feels weird.  We already have an `Exception` from the `Try...()` and we're stealing the message and throwing it away to create a new one -- just to get the `WriteError()` call.  

Won'd this also throw away the original throw context (stack info and etc)?

Maybe we should just manually do the `WriteError()` here and let it return the original exception.